torchvision/prototype/transforms/_color.py

```diff
@@ -120,5 +121,70 @@ def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
 
         for transform in params["image_transforms"]:
             input = transform(input)
+        return input
+
+
+class _RandomChannelShuffle(Transform):
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        image = query_image(sample)
+        num_channels, _, _ = get_image_dimensions(image)
+        return dict(permutation=torch.randperm(num_channels))
+
+    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
+        if not (isinstance(input, (features.Image, PIL.Image.Image)) or is_simple_tensor(input)):
```

Comment:
The structure is different from the usual if / elif / else here, because we use the same "kernel" for all supported types. If we have dedicated kernels for this as proposed above, we can go back to the regular structure.